microbench/src/main/java/io/netty/microbench/http2/internal/hpack/HeadersSize.java

```diff
@@ -56,7 +59,7 @@
         return Header.createHeaders(numHeaders, nameLength, valueLength, limitAscii);
     }
 
-    public ByteArrayOutputStream newOutputStream() {
-        return new ByteArrayOutputStream(numHeaders * (nameLength + valueLength));
+    public ByteBuf newOutBuffer() {
+        return Unpooled.buffer(numHeaders * (nameLength + valueLength));
```

Comment:
I'm not sure if we really use this benchmark any more ... but should we include an `ByteBufAllocator` in the mix?